app/src/main/java/com/arygm/quickfix/model/category/CategoryRepository.kt

```diff
@@ -4,15 +4,15 @@ interface CategoryRepository {
 
   fun init(onSuccess: () -> Unit)
 
-  fun fetchCategories(onSuccess: (List<Category?>) -> Unit, onFailure: (Exception) -> Unit)
+  suspend fun fetchCategories(onSuccess: (List<Category?>) -> Unit, onFailure: (Exception) -> Unit)
```

Comment:
Can you explain what a suspend fun is?

Reply:
A `suspend` function is a function that can be paused and resumed later. It is designed to work with **coroutines**, which are a powerful tool for handling asynchronous or long-running operations in Kotlin.

When a `suspend` function is called, it doesn't block the calling thread. Instead, it suspends execution and allows other code to run until the function resumes. 

By all means don't hesitate to ask if anything else needs clarification 😄